python/ray/autoscaler/v2/instance_manager/reconciler.py

```diff
@@ -601,3 +621,129 @@ def _reconciled_im_status_from_ray_status(
             return cur_im_status
 
         return reconciled_im_status
+
+    @staticmethod
+    def _handle_instances_launch(
+        instance_manager: InstanceManager, autoscaling_config: AutoscalingConfig
+    ):
+
+        instances, version = Reconciler._get_im_instances(instance_manager)
+
+        queued_instances = []
+        requested_instances = []
+        allocated_instances = []
+
+        for instance in instances:
+            if instance.status == IMInstance.QUEUED:
+                queued_instances.append(instance)
+            elif instance.status == IMInstance.REQUESTED:
+                requested_instances.append(instance)
+            elif instance.cloud_instance_id:
+                allocated_instances.append(instance)
+
+        if not queued_instances:
+            # No QUEUED instances
```

Comment:
When do we add QUEUED?

Reply:
we will add QUEUED when we scale up the cluster (after consulting the scheduler), and this process is before this.